app/scripts/controllers/deployment.js

```diff
@@ -223,190 +155,80 @@ angular.module('openshiftConsole')
-            // After the deployment ends remove him from the deploymentConfigDeploymentsInProgress structure.
-            var status = $filter('deploymentStatus')(deployment);
-            if (status === "Complete" || status === "Failed"){
-              delete $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName][deploymentName];
-            }
-          }
-
-          // Extract the causes from the encoded deployment config
-          if (deployment) {
-            if (action !== "DELETED") {
-              deployment.causes = $filter('deploymentCauses')(deployment);
-            }
-          }
-          else {
-            angular.forEach($scope.deployments, function(deployment) {
-              deployment.causes = $filter('deploymentCauses')(deployment);
-            });
-          }
-        }));
+        // List limit ranges in this project to determine if there is a default
+        // CPU request for autoscaling.
+        DataService.list("limitranges", context, function(response) {
+          limitRanges = response.by("metadata.name");
+          updateHPAWarnings();
+        });
 
-        // Sets up subscription for imageStreams
         watches.push(DataService.watch("imagestreams", context, function(imageStreams) {
           $scope.imageStreams = imageStreams.by("metadata.name");
           ImageStreamResolver.buildDockerRefMapForImageStreams($scope.imageStreams, $scope.imageStreamImageRefByDockerReference);
-          ImageStreamResolver.fetchReferencedImageStreamImages($scope.podTemplates, $scope.imagesByDockerReference, $scope.imageStreamImageRefByDockerReference, context);
+          // If the dep config has been loaded already
+          if ($scope.deployment) {
+            ImageStreamResolver.fetchReferencedImageStreamImages([$scope.deployment.spec.template], $scope.imagesByDockerReference, $scope.imageStreamImageRefByDockerReference, context);
+          }
           Logger.log("imagestreams (subscribe)", $scope.imageStreams);
         }));
 
-        watches.push(DataService.watch("builds", context, function(builds) {
-          $scope.builds = builds.by("metadata.name");
-          Logger.log("builds (subscribe)", $scope.builds);
-        }));
-
         watches.push(DataService.watch({
           group: "extensions",
           resource: "horizontalpodautoscalers"
-        }, context, function(data) {
-          allHPA = data.by("metadata.name");
-          updateHPA();
-          updateHPAWarnings();
-        }, {poll: limitWatches, pollInterval: 60 * 1000}));
-
-        // List limit ranges in this project to determine if there is a default
-        // CPU request for autoscaling.
-        DataService.list("limitranges", context, function(response) {
-          limitRanges = response.by("metadata.name");
+        }, context, function(hpa) {
+          $scope.autoscalers =
+            HPAService.filterHPA(hpa.by("metadata.name"), 'Deployment', $routeParams.deployment);
```

Comment:
does filterHPA handle the deployment being null in case this fires before deployment is loaded